lib/client/db/postgres/repl/repl.go

```diff
@@ -0,0 +1,258 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package repl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgconn"
+	"golang.org/x/term"
+
+	"github.com/gravitational/teleport"
+	clientproto "github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/asciitable"
+)
+
+type REPL struct {
+	ctx        context.Context
+	cancelFunc context.CancelCauseFunc
+	conn       *pgconn.PgConn
+	client     io.ReadWriter
+	serverConn net.Conn
+	route      clientproto.RouteToDatabase
+	term       *term.Terminal
+	commands   map[string]*command
+}
+
+func Start(ctx context.Context, client io.ReadWriter, serverConn net.Conn, route clientproto.RouteToDatabase) (*REPL, error) {
+	config, err := pgconn.ParseConfig(fmt.Sprintf("postgres://%s@%s/%s", route.Username, hostnamePlaceholder, route.Database))
```

Comment:
maybe add application name and connection timeout?
nit: also, it might be easier to express/read in DSN instead of URL.